extensions/AzureOpenAI/AzureOpenAIBulkTextEmbeddingGenerator.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.OpenAI;
+using Azure.Identity;
+using Microsoft.Extensions.Logging;
+using Microsoft.KernelMemory.AI.OpenAI;
+using Microsoft.KernelMemory.Diagnostics;
+
+namespace Microsoft.KernelMemory.AI.AzureOpenAI;
+
+[Experimental("KMEXP01")]
+public sealed class AzureOpenAIBulkTextEmbeddingGenerator : IBulkTextEmbeddingGenerator
```

Comment:
I would add this interface to the existing `AzureOpenAITextEmbeddingGenerator` class:

```csharp
public sealed class AzureOpenAITextEmbeddingGenerator : ITextEmbeddingGenerator, IBulkTextEmbeddingGenerator
```

so we can avoid duplicating the rest of the class. Then at runtime we can check is `IBulkTextEmbeddingGenerator` is available.

Reply:
Ok sounds good, I'll be OOF for two days, then I'll implement all suggestions